### PR TITLE
feat: ✨ Add Leave Lobby Functionality 

### LINKED ISSUE #39: Allow Catniss to leave the lobby
Leave Lobby button for Catniss to click on to take them back to the main page in case she does not want to play in the current lobby.

Catniss can leave the lobby by closing the tab so that she is not forced to press the "leave lobby" button.

This card does not include the case where the browser crashes or runs out of battery

Task List

- [x] Add a leave lobby button to the main lobby page
- [x] Closing the tab should perform the same actions as the button
- [x] Clicking the leave lobby button should remove you from the lobby entirely
- [x] If you are the host then make another player the host 
- [x] If you are the last player in the lobby the lobby should be deleted 
- [x] Create a pop-up that will display if the player tries to leave an in progress game

Acceptance Criteria
- Cantiss can leave a lobby on the lobby page
- Clicking on button takes Catniss back to main page
- Catniss cannot leave while a game is in progress
- If Catniss is the host then another player will be the host so that the lobby does not close when the host leaves.
- if Catniss was the last player in the lobby the lobby should be deleted so that there is not a large sum of empty lobbies constantly running.
- if the game is in progress and the user tries to close the tab it will ask if are user using the [beforeunload](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event)
- if the server side throws an error the client should display it (make this work for start game as well)

### PR DESCRIPTION
**Pull request by mrjoshua520**
closes #39

### CODE REVIEW

**Comment by nstringham** [comment]
while playing the game if you reload the page then you leave the lobby. is that acceptable behavior?

File: client/src/components/Lobby.svelte
Comment (nstringham) [inline client/src/components/Lobby.svelte]: I doubt you intended to leave that in

File: client/src/components/Lobby.svelte
Comment (nstringham) [inline client/src/components/Lobby.svelte]: I feel like it would look better if it was like
```svelte
{#if auth.currentUser?.uid === lobby.uids[0]}
  <div class="start">
    <Button on:click={() => startGame()}><Label>Start Game</Label></Button>
  </div>
{/if}
```

File: client/src/components/Lobby.svelte
Comment (nstringham) [inline client/src/components/Lobby.svelte:75]: this makes it go to the home page before it is done leaving the lobby because leaveLobby is not awaited

File: client/src/components/Lobby.svelte
Comment (nstringham) [inline client/src/components/Lobby.svelte]: you probobly don't want 2 things with the same class like that

File: client/src/components/Lobby.svelte
Comment (nstringham) [inline client/src/components/Lobby.svelte]: if these 3 classes are the same then you should only have one class that all 3 elements share

File: client/src/lib/firebase/start-lobby.ts
Comment (nstringham) [inline client/src/lib/firebase/start-lobby.ts]: don't really need this but whatever

File: client/src/lib/firebase/leave-lobby.ts
Comment (nstringham) [inline client/src/lib/firebase/leave-lobby.ts]: don't really need this but whatever

File: client/src/routes/game/+page.svelte
Comment (nstringham) [inline client/src/routes/game/+page.svelte]: this code always runs even if the user clicks "cancel" maybe use the "unload" event

**Review by nstringham — CHANGES_REQUESTED** [review]

File: client/src/routes/game/+page.svelte
Comment (mrjoshua520) [reply client/src/routes/game/+page.svelte]: We decided not to have the user leave during the unload event so this won't need to be addressed

File: functions/src/lobby.ts
Comment (nstringham) [inline functions/src/lobby.ts:174]: this await doesn't do anything because `transaction.delete` does not return a promise

Comment (nstringham) [inline functions/src/lobby.ts]: this await doesn't do anything because `transaction.delete` does not return a promise

File: functions/src/lobby.ts
Comment (nstringham) [inline functions/src/lobby.ts]: this await doesn't do anything because `transaction.update` does not return a promise

File: client/src/components/Lobby.svelte
Comment (nstringham) [inline client/src/components/Lobby.svelte]: I don't like how the `<div>` is on the same line as `{#if}` but the `</div>` is on a diffrent line from `{/if}`

**Review by nstringham — CHANGES_REQUESTED** [review]

**Review by nstringham — DISMISSED** [review]

**Review by nstringham — APPROVED** [review]
approving again after resolving merge conflicts

**Pull request merged**
